Sources/SwiftProtobuf/Message.swift

```diff
@@ -112,17 +155,33 @@ public protocol Message: CustomDebugStringConvertible {
   // They are implemented in the protocol, not in the generated structs,
   // so can be overridden in user code by defining custom extensions to
   // the generated struct.
+
+  /// Hash value generated from this message's contents, for conformance with
+  /// the `Hashable` protocol
   var hashValue: Int { get }
+
+  /// Textual representation of this message's contents suitable for debugging,
+  /// for conformance with the `CustomDebutStringConvertible` protocol.
   var debugDescription: String { get }
 }
 
 public extension Message {
+
+  /// Generates a hash based on the message's full contents. Can be overridden
+  /// to improve performance and/or remove some values from being used for the
+  /// hash.
+  ///
+  /// If you override this, make sure you maintain the property that values
+  /// which are `==` to each other have identical `hashValues`, providing a
+  /// custom implementation of `==` if necessary.
   var hashValue: Int {
     let visitor = HashVisitor()
     try? traverse(visitor: visitor)
     return visitor.hashValue
   }
 
+  /// Uses a `Visitor` to recursively generate a description. May be overridden
```

Comment:
The fact that a `Visitor` is used here is an unimportant detail—it doesn't need to be included (and the implementation of this function is changing soon anyway with a PR from @thomasvl).